pgmpy/models/LinearGaussianBayesianNetwork.py

```diff
@@ -272,18 +290,126 @@ def simulate(self, n_samples=1000, seed=None):
         >>> model.add_cpds(cpd1, cpd2, cpd3)
         >>> model.simulate(n_samples=500, seed=42)
         """
+        evidence = {} if evidence is None else evidence
```

Comment:
Can you write comments with the main steps of the method? We usually do comments such as: `Step 1: Check all the arguemnts`, `Step 2: If do variables are specified, ... `. Have a look at other algorithm implementations for examples.